tests/testStreamOpen.cpp

```diff
@@ -0,0 +1,73 @@
+#include <oboe/Oboe.h>/*
+ * Copyright 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <oboe/Oboe.h>
+
+using namespace oboe;
+
+
+class StreamOpen : public ::testing::Test {
+
+protected:
+
+    bool openStream(){
+        Result r = mBuilder.openStream(&mStream);
+        EXPECT_EQ(r, Result::OK) << "Failed to open stream " << convertToText(r);
+        return (r == Result::OK);
+    }
+
+    void closeStream(){
+        Result r = mStream->close();
+        if (r != Result::OK){
+            FAIL() << "Failed to close stream. " << convertToText(r);
+        }
+    }
+
+    AudioStreamBuilder mBuilder;
+    AudioStream *mStream = nullptr;
+
+};
+
+TEST_F(StreamOpen, ForOpenSLESDefaultSampleRateIsUsed){
+
+    DefaultStreamValues::SampleRate = 44100;
+    openStream();
+    if (mStream->getAudioApi() == AudioApi::OpenSLES){
```

Comment:
Should request OpenSL ES when you open.

Reply:
Of course! That simplifies each test nicely.